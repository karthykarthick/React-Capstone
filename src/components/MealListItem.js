import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { truncate } from '../helpers';

const MealListItem = ({ meal }) => (
  <Link
    className="meal"
    to={`/meal/${meal.id}`}
  >
    <img
      className="meal__image"
      src={meal.image}
      alt={meal.title}
    />
    <div className="meal__info">
      <h4 className="meal__info__title">{truncate(meal.title, 35)}</h4>
      <span className="meal__info__category">{meal.category}</span>
    </div>
  </Link>
);

MealListItem.propTypes = {
  meal: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default MealListItem;
