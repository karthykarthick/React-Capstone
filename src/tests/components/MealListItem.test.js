import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import MealListItem from '../../components/MealListItem';

const mealTest = {
  id: '234214',
  title: 'This is a title',
  image: 'https://www.google.com/images/test.png',
  category: 'Seafood',
};

const renderer = new ShallowRenderer();

it('should render MealListItem correctly', () => {
  renderer.render(<MealListItem meal={mealTest} />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('should render MealListItem correctly with dots on title', () => {
  const newMeal = {
    ...mealTest,
    title: 'This is a really long title that should be trunctated',
  };

  renderer.render(<MealListItem meal={newMeal} />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
