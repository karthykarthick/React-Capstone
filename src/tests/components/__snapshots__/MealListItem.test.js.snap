// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MealListItem correctly 1`] = `
<Link
  className="meal"
  to="/meal/234214"
>
  <img
    alt="This is a title"
    className="meal__image"
    src="https://www.google.com/images/test.png"
  />
  <div
    className="meal__info"
  >
    <h4
      className="meal__info__title"
    >
      This is a title
    </h4>
    <span
      className="meal__info__category"
    >
      Seafood
    </span>
  </div>
</Link>
`;

exports[`should render MealListItem correctly with dots on title 1`] = `
<Link
  className="meal"
  to="/meal/234214"
>
  <img
    alt="This is a really long title that should be trunctated"
    className="meal__image"
    src="https://www.google.com/images/test.png"
  />
  <div
    className="meal__info"
  >
    <h4
      className="meal__info__title"
    >
      This is a really long title that s...
    </h4>
    <span
      className="meal__info__category"
    >
      Seafood
    </span>
  </div>
</Link>
`;
